### PR TITLE
feat(azure): Add support for Azure HDInsight

### PR DESCRIPTION
**Pull request by ekifel**
Issue #670 

Details:
Skipped `azurerm_hdinsight_storm_cluster`, `azurerm_hdinsight_ml_services_cluster`, `azurerm_hdinsight_rserver_cluster` resources as they will be removed in the next major version of the AzureRM Terraform Provider.

### CODE REVIEW

**Comment by ekifel** [comment]
@alikhajeh1, It is ready for review! :)

File: internal/providers/terraform/azure/testdata/hdinsight_kafka_cluster_test/hdinsight_kafka_cluster_test.golden [hunk 1]
```diff
@@ -0,0 +1,33 @@
+
+ Name                                            Monthly Qty  Unit                  Monthly Cost 
+                                                                                                 
+ azurerm_hdinsight_kafka_cluster.non_usage                                                       
+ ├─ Head node (NC24)                                   1,460  hours                    $7,478.12 
+ ├─ Worker node (D3)                                     730  hours                      $215.35 
+ ├─ Zookeeper node (D3)                                2,190  hours                      $646.05 
+ ├─ Managed OS disks                                       3  months                     $446.04 
+ └─ Disk operations                         Cost depends on usage: $0.00036 per 100K operations
```
Comment (tim775) [inline internal/providers/terraform/azure/testdata/hdinsight_kafka_cluster_test/hdinsight_kafka_cluster_test.golden:9]: Need to rebase and regenerate so you pick up the "Monthly cost depends..." change.

**Comment by tim775** [comment]
@ekifel Needs a rebase but otherwise looks good.

**Comment by ekifel** [comment]
@tim775, thanks! Rebased

**Comment by tim775** [comment]
@alikhajeh1 lgtm

**Pull request merged**